sdk/dotnet/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -160,10 +160,14 @@ private static void ThrowAliasPropertyConflict(string name)
                         }
 
                         var aliasSpec = new Pulumirpc.Alias.Types.Spec();
-                        aliasSpec.Name = await MaybeAwait(aliasVal.Name);
-                        aliasSpec.Type = await MaybeAwait(aliasVal.Type);
-                        aliasSpec.Stack= await MaybeAwait(aliasVal.Stack);
-                        aliasSpec.Project = await MaybeAwait(aliasVal.Project);
+                        var aliasName = await MaybeAwait(aliasVal.Name);
+                        if (aliasName != null) { aliasSpec.Name = aliasName; }
+                        var aliasType = await MaybeAwait(aliasVal.Type);
+                        if (aliasType != null) { aliasSpec.Type = aliasType; }
+                        var aliasStack= await MaybeAwait(aliasVal.Stack);
+                        if (aliasStack != null) { aliasSpec.Stack = aliasStack; }
+                        var aliasProject = await MaybeAwait(aliasVal.Project);
+                        if (aliasProject != null) { aliasSpec.Project = aliasProject; }
```

Comment:
What happens if an old SDK without these changes runs against the new engine?

Reply:
It will send the old `repeated string` of urns and the engine will accept that. The engine will in fact accept both the old string urns and new alias specs in the same request and just concatenate them all together.